x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/edit_field/edit_field.tsx

```diff
@@ -199,7 +227,7 @@ export const EditField = React.memo(
                 fill
                 onClick={submitForm}
                 type="submit"
-                disabled={form.isSubmitted && !form.isValid}
+                disabled={(form.isSubmitted && !form.isValid) || !isFormModified}
```

Comment:
Unrelated to this PR changes: the condition checks for `form.isSubmitted && !form.isValid` so the form lets the user click "Update" and then shows errors. I wonder if we should disable the button is form is just not valid 🤔

Reply:
Yes, I also noticed this behaviour that is not very intuitive. I did a small test and removed the `form.isSubmitted` clause but then the Update button doesn't work as expected - it's always disabled regardless of whether the form is valid or not. Probably the `form.isSubmitted` clause was added to solve this issue, but I think it might be better if we fix it properly so that the Update button is disabled when the form is not valid.

Reply:
This is a UX behaviour we've implemented in many forms (it was not added to solve an issue).

The reasoning is:  _while_ a user is updating a form it is still neither valid nor invalid. It is in the process of being filled. Only when the user clicks the "send" button (they think they got it all right) we validate and mark the form state as valid or invalid.

It has the benefit of **not** showing a disabled button as soon as the form is rendered. In large forms there is no clear way to answer the question "_why is the send button disabled?_". If a field is required, the only clue for the user is to search for "*" (which we agree is, by definition, not very "clear" 😊 ) the user find himself touching each field until they find which one is failing. They might even not do that and give up.

By having the "send" button always active until the user first attempts to submit it we solve that UX problem. After clicking the button, if there are errors we will show a very clear red callout to indicate failing fields.

I've commented this in multiple places across different issues/SDH so maybe I am the only one who see this UX problem. 😄